extensions/cornerstone/src/panels/PanelSegmentation.tsx

```diff
@@ -168,27 +172,29 @@ export default function PanelSegmentation({
     };
   });
 
-  // Update the map of last selected segmentation IDs for the active viewport id.
+  // Update the store of last selected segmentation IDs for the active viewport id.
   const activeSegmentationInfo = segmentationsWithRepresentations.find(
     info => info.representation?.active
   );
 
-  if (activeSegmentationInfo) {
-    const typeToSegmentationIdMap =
-      selectedSegmentationIdByViewportAndType.get(activeViewportId) ?? new Map<string, string>();
+  const { selectedSegmentationsForViewport, setSelectedSegmentationsForViewport } =
+    useSelectedSegmentationsForViewportStore.getState();
```

Comment:
I don't think you need to do getState in a react component, getState is useful for non react. I think there are other api for in react, this works, but is not reactive